File: packages/common/models/chat/chat-message.ts
import { WithIdOf } from "../../types/id";
import { PublicUser } from "../user";
import { Channel, UserChannel } from "./channel";

export interface ChatMessage {
    sender: PublicUser;
    content: string;
    date: Date;
}

export interface ChannelMessage extends WithIdOf<Channel> {
    message: ChatMessage;
}

export interface ChatHistoryMessage extends UserChannel {
    idMessage: number;
    date: Pick<ChatMessage, 'date'>;
    content: Pick<ChatMessage, 'content'>;
}

File: packages/mobile/lib/classes/chat-message.dart
import 'package:mobile/classes/user.dart';

class ChatMessage {
  final PublicUser sender;
  final String content;
  final String date;

  ChatMessage(
      {required this.sender, required this.content, required this.date});

  factory ChatMessage.fromJson(Map<String, dynamic> json) {
    return ChatMessage(
      sender: PublicUser.fromJson(json['sender']),
      content: json['content'] as String,
      date: json['date'] as String,
    );
  }

  Map<String, dynamic> toJson() => {
        "sender": sender,
        "content": content,
        "date": date,
      };
}
